tests/unittests/algo/test_tpe.py

```diff
@@ -0,0 +1,394 @@
+
+from orion.algo.space import Integer, Real, Space
+from orion.algo.tpe import adaptive_parzen_estimator, compute_max_ei_point, GMMSampler, TPE
+
+
+@pytest.fixture()
+def space():
+    """Return an optimization space"""
+    space = Space()
+    dim1 = Real('yolo1', 'uniform', -10, 20)
+    space.register(dim1)
+    dim2 = Real('yolo2', 'uniform', -5, 10)
+    space.register(dim2)
+
+    return space
+
+
+@pytest.fixture
+def tpe(space):
+    """Return an instance of TPE."""
+    return TPE(space)
+
+
+def test_compute_max_ei_point():
+    """Test that max ei point is computed correctly"""
+    points = numpy.linspace(-3, 3, num=10)
+    below_likelis = numpy.linspace(0.5, 0.9, num=10)
+    above_likes = numpy.linspace(0.2, 0.5, num=10)
+
+    numpy.random.shuffle(below_likelis)
+    numpy.random.shuffle(above_likes)
+    max_ei_index = (below_likelis - above_likes).argmax()
+
+    max_ei_point = compute_max_ei_point(points, below_likelis, above_likes)
+    assert max_ei_point == points[max_ei_index]
+
+
+def test_adaptive_parzen_normal_estimator():
+    """Test adaptive parzen estimator"""
+    low = -1
+    high = 5
+
+    obs_mus = [1.2]
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=1.0,
+                                                     equal_weight=False, flat_num=25)
+    assert list(mus) == [1.2, 2]
+    assert list(sigmas) == [3, 6]
+    assert list(weights) == [1.0 / 2, 1.0 / 2]
+
+    obs_mus = [3.4]
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=0.5,
+                                                     equal_weight=False, flat_num=25)
+    assert list(mus) == [2, 3.4]
+    assert list(sigmas) == [6, 3]
+    assert list(weights) == [0.5 / 2, 1.0 / 2]
+
+    obs_mus = numpy.linspace(-1, 5, num=30, endpoint=False)
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=1.0,
+                                                     equal_weight=False, flat_num=25)
+
+    assert len(mus) == len(sigmas) == len(weights) == 30 + 1
+    assert numpy.all(weights[-25:] == 1 / 31)
+    assert numpy.all(sigmas == 6 / 10)
+
+
+def test_adaptive_parzen_normal_estimator_weight():
+    """Test the weight for the normal components"""
+    obs_mus = numpy.linspace(-1, 5, num=30, endpoint=False)
+    low = -1
+    high = 5
+
+    # equal weight
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=1.0,
+                                                     equal_weight=True, flat_num=25)
+    assert numpy.all(weights == 1 / 31)
+    assert numpy.all(sigmas == 6 / 10)
+
+    # prior weight
+    mus, sigmas, weights = adaptive_parzen_estimator(obs_mus, low, high, prior_weight=0.5,
+                                                     equal_weight=False, flat_num=25)
+    prior_pos = numpy.searchsorted(mus, 2)
+    assert weights[prior_pos] == 0.5 / 31
+    assert numpy.all(weights[31 - prior_pos:] == weights[-1])
```

Comment:
The linear part should be tested as well, otherwise this test would pass even if equal_weight=False would return the same than equal_weight=True.

Reply:
good idea